### ISSUE TITLE
Layer Cache not found

### ISSUE THREAD

**Issue by NoamDev** [satackey/action-docker-layer-caching#19]
I use the action with the following code in order to make sure that PRs' cache do not affect master:
``` yaml
    - uses: satackey/action-docker-layer-caching@6b09a11416d285a6bf2a9d1ce2484c878f7c985e
      with:
        key: dlc_${{ github.ref }}_{hash}
        restore-keys: |
          dlc_refs/heads/master_
          dlc_${{ github.ref }}_
```
However, when the action run on a new PR, I get the following error:
```
Run satackey/action-docker-layer-caching@6b09a11416d285a6bf2a9d1ce2484c878f7c985e
  with:
    key: dlc_refs/pull/32/merge_{hash}
    restore-keys: dlc_refs/heads/master_
  dlc_refs/pull/32/merge_
  
    concurrency: 4
Received 14363 of 14363 (100.0%), 0.7 MBs/sec
Cache Size: ~0 MB (14363 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/6c9bf4f4-3bb5-420a-9886-8bdb81eed2d9/cache.tzst -P -C /home/runner/work/ridewithto_cloudrun/ridewithto_cloudrun
Error: Layer cache not found: {"id":"bfc566732015e74554c940abfdedb5ca215be6a392cad0f189b171298527cc09"}
##[error]Error: Layer cache not found: {"id":"bfc566732015e74554c940abfdedb5ca215be6a392cad0f189b171298527cc09"}
    at LayerCache.restoreSingleLayerBy (/home/runner/work/_actions/satackey/action-docker-layer-caching/6b09a11416d285a6bf2a9d1ce2484c878f7c985e/dist/ttsc-dist/main.js/index.js:33173:19)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Promise.all (index 1)
    at async LayerCache.restoreLayers (/home/runner/work/_actions/satackey/action-docker-layer-caching/6b09a11416d285a6bf2a9d1ce2484c878f7c985e/dist/ttsc-dist/main.js/index.js:33162:58)
    at async LayerCache.restore (/home/runner/work/_actions/satackey/action-docker-layer-caching/6b09a11416d285a6bf2a9d1ce2484c878f7c985e/dist/ttsc-dist/main.js/index.js:33140:42)
    at async main (/home/runner/work/_actions/satackey/action-docker-layer-caching/6b09a11416d285a6bf2a9d1ce2484c878f7c985e/dist/ttsc-dist/main.js/index.js:43988:
25)
```
Changing the key prefix from dlc to something works, but it's not a practical workaround..

**Comment by NoamDev** [comment]
I've investigated it a bit further, the problem is, I think, that while dlc_refs/heads/master_ cache exists, dlc_${{ github.ref }}_ does not.
So it restores root with  dlc_refs/heads/master_ cache but then tries to restore the specific layer with `key` i.e. dlc_${{ github.ref }}_ and obviously fails.

As seen here:
https://github.com/satackey/action-docker-layer-caching/blob/0c6d193ecc434135211fc290d7d3b32d7a0d1cf2/src/LayerCache.ts#L202-L213
restoring is being done with the store key, it should instead use the same restore key which the root was restored from.

**Comment by NoamDev** [comment]
So, I believe the main problem is that single leyer restoration do not use the restore-keys.

But mapping restore keys to layer specific restore key is a bit complicated, because they do not include "{hash}"..

What do you think?

**Comment by djarecka** [comment]
I have a similar issue when running `satackey/action-docker-layer-caching@v0.0.5`. It doesn't fail every single time, but quite often.

**Comment by benmezger** [comment]
Been having the same issue over here: https://github.com/benmezger/dotfiles/runs/975814502?check_suite_focus=true

**Comment by satackey** [comment]
Thank you for your report.

>  restoring is being done with the store key, it should instead use the same restore key which the root was restored from.

You're right. I'll fix this shortly.

> So, I believe the main problem is that single leyer restoration do not use the restore-keys.
> But mapping restore keys to layer specific restore key is a bit complicated, because they do not include "{hash}"..
> What do you think?

Currently the hash of root key is created by the image list to be cached.
I'm going to reduce the complication by using the hash of the metadata file in the root cache as the key.
This allows you to find the key before the `{hash}` is replaced at restoring step.
Thus, the layer id stored in the root cache metadata can be used as a layer cache key, even when using a different `restore-key` than the `key`.

**Comment by satackey** [comment]
I have reproduced this in [this result](https://github.com/satackey/action-docker-layer-caching/runs/982683419?check_suite_focus=true#step:6:140) and fixed it in [v0.0.7](https://github.com/satackey/action-docker-layer-caching/releases/tag/v0.0.7) (#22).

Please check in your environment that this version works successfully.

**Comment by NoamDev** [comment]
@satackey Nice!
Should I use a new cache key for the new version? (I guess it's not compatible with my existing cache)

**Comment by djarecka** [comment]
works for me! Thank you!

**Comment by satackey** [comment]
@NoamDev
As you say, it's not compatible with the existing cache, but I've made some changes below so you don't have to.

I found that in restoring using `@actions/cache` library, the cache was not found for different file paths, even if the keys were the same.

I changed the internal temporary path used for the image cache to include the tag or hash specified in the `uses`.
Thus your existing cache will not be restored with the new version.
